cms/src/main/java/uk/nhs/digital/externalstorage/workflow/AbstractExternalFileTask.java

```diff
@@ -0,0 +1,99 @@
+package uk.nhs.digital.externalstorage.workflow;
+
+import org.hippoecm.frontend.session.UserSession;
+import org.hippoecm.repository.api.WorkflowException;
+import org.onehippo.cms7.services.HippoServiceRegistry;
+import org.onehippo.cms7.services.eventbus.HippoEventBus;
+import org.onehippo.repository.documentworkflow.DocumentVariant;
+import org.onehippo.repository.documentworkflow.task.AbstractDocumentTask;
+import org.onehippo.repository.events.HippoWorkflowEvent;
+import uk.nhs.digital.externalstorage.ExternalStorageConstants;
+import uk.nhs.digital.externalstorage.s3.S3Connector;
+
+import java.rmi.RemoteException;
+import javax.jcr.Node;
+import javax.jcr.NodeIterator;
+import javax.jcr.RepositoryException;
+import javax.jcr.query.Query;
+import javax.jcr.query.QueryResult;
+
+public abstract class AbstractExternalFileTask extends AbstractDocumentTask {
+
+    private String variantState;
+
+    protected S3Connector s3Connector;
```

Comment:
Can you make s3 connector private as well? You could pass it into the setTargetState method but if all you do is call HippoServiceRegistry.getService(S3Connector.class) you don't even need to set this on the abstract class really. you could just get it in each task when you need it. I don't mind either way as long as the local variables are private.